Cod3rsGrowth.Web/wwwroot/controller/CriarEditarEmpresas.controller.js

```diff
@@ -60,24 +77,67 @@ sap.ui.define([
 				dataAbertura: modelo.getProperty(NOME_PROPRIEDADE_DATA_ABERTURA_EMPRESA),
-				naturezaJuridica: modelo.getProperty(NOME_PROPRIEDADE_NATUREZA_JURIDICA_EMPRESA),
-				porte: modelo.getProperty(NOME_PROPRIEDADE_PORTE_EMPRESA),
-				matrizFilial: modelo.getProperty(NOME_PROPRIEDADE_MATRIZ_FILIAL_EMPRESA),
-				capitalSocial: modelo.getProperty(NOME_PROPRIEDADE_CAPITAL_SOCIAL_EMPRESA),
-				idEndereco: 0
+				naturezaJuridica: parseInt(modelo.getProperty(NOME_PROPRIEDADE_NATUREZA_JURIDICA_EMPRESA)),
+				porte: parseInt(modelo.getProperty(NOME_PROPRIEDADE_PORTE_EMPRESA)),
+				matrizFilial: parseInt(modelo.getProperty(NOME_PROPRIEDADE_MATRIZ_FILIAL_EMPRESA)),
+				capitalSocial: parseInt(modelo.getProperty(NOME_PROPRIEDADE_CAPITAL_SOCIAL_EMPRESA))
 			}
 		},
+		_populaTelaComValoresEmpresaEditar: async function (idEmpresaAtualizar) {	
+			this._idEmpresaAtualizar = idEmpresaAtualizar;
+			const modelo = this.getView().getModel();
+			const empresaEditar = await ServicoEmpresas.
+				obterEmpresaPorId(idEmpresaAtualizar);
+			modelo.setProperty(NOME_PROPRIEDADE_RAZAO_SOCIAL_EMPRESA,
+				empresaEditar.razaoSocial);
+			modelo.setProperty(NOME_PROPRIEDADE_NOME_FANTASIA_EMPRESA,
+				empresaEditar.nomeFantasia);
+			modelo.setProperty(NOME_PROPRIEDADE_CNPJ_EMPRESA,
+				empresaEditar.cnpj);
+			modelo.setProperty(NOME_PROPRIEDADE_SITUACAO_CADASTRAL_EMPRESA,
+				empresaEditar.situacaoCadastral ? 1 : 0);
+			modelo.setProperty(NOME_PROPRIEDADE_DATA_ABERTURA_EMPRESA,
+				empresaEditar.dataAbertura);
+			modelo.setProperty(NOME_PROPRIEDADE_NATUREZA_JURIDICA_EMPRESA,
+				empresaEditar.naturezaJuridica);
+			modelo.setProperty(NOME_PROPRIEDADE_PORTE_EMPRESA,
+				empresaEditar.porte);
+			modelo.setProperty(NOME_PROPRIEDADE_MATRIZ_FILIAL_EMPRESA,
+				empresaEditar.matrizFilial);
+			modelo.setProperty(NOME_PROPRIEDADE_CAPITAL_SOCIAL_EMPRESA,
+				empresaEditar.capitalSocial);
+			await this._populaTelaComValoresEnderecoDaEmpresaEditar(empresaEditar.idEndereco);
+		},
+		_populaTelaComValoresEnderecoDaEmpresaEditar: async function (id) {
+			const modelo = this.getView().getModel();
+			const enderecoEmpresaEditar = await ServicoEnderecos.obterEnderecoPorId(id);
+			this._idEnderecoAtualizar = enderecoEmpresaEditar.id;
+			modelo.setProperty(NOME_PROPRIEDADE_NUMERO_EMPRESA,
+				enderecoEmpresaEditar.numero);
+			modelo.setProperty(NOME_PROPRIEDADE_CEP_EMPRESA,
+				enderecoEmpresaEditar.cep);
+			modelo.setProperty(NOME_PROPRIEDADE_MUNICIPIO_EMPRESA,
+				enderecoEmpresaEditar.municipio);
+			modelo.setProperty(NOME_PROPRIEDADE_BAIRRO_EMPRESA,
+				enderecoEmpresaEditar.bairro);
+			modelo.setProperty(NOME_PROPRIEDADE_RUA_EMPRESA,
+				enderecoEmpresaEditar.rua);
+			modelo.setProperty(NOME_PROPRIEDADE_COMPLEMENTO_EMPRESA,
+				enderecoEmpresaEditar.complemento);
+			modelo.setProperty(NOME_PROPRIEDADE_ESTADO_EMPRESA,
+				enderecoEmpresaEditar.estado);
+		},
 		_retornaValoresEndereco() {
-			
 			const modelo = this.getView().getModel();
```

Comment:
nao tem necessidade de pegar uma por uma utiliza o this.getView().getModel().getdata()